src/onboarding.ts

```diff
@@ -228,14 +229,79 @@ promptfoo eval
 Afterwards, you can view the results by running \`promptfoo view\`
 `;
 
-function recordOnboardingStep(step: string, properties: Record<string, EventValue> = {}) {
+function recordOnboardingStep(step: string, properties: any = {}) {
   telemetry.recordAndSend('funnel', {
     type: 'eval onboarding',
     step,
     ...properties,
   });
 }
 
+/**
+ * Iterate through user choices and determine if the user has selected a provider that needs an API key
+ * but has not set and API key in their enviorment.
+ */
+export function reportProviderAPIKeyWarnings(providerChoices: (string | object)[]): string[] {
```

Comment:
Great idea to display a warning for this!

Is there a way we can simplify this function? e.g. using the map more directly and/or reducing the switches/branching

Reply:
Updated to address comments and reduce complexity -- fresh morning eyes always help lol 👀